test/t/test_python.py

```diff
@@ -45,3 +45,23 @@ def test_9(self, completion):
     )
     def test_bb(self, completion):
         assert "-bb" in completion
+
+    @pytest.mark.complete("python -m foo bar -p ", cwd="python")
+    def test_script_args_1(self, completion):
+        assert "bar.txt" in completion
+
+    @pytest.mark.complete("python -mfoo bar -p ", cwd="python")
+    def test_script_args_2(self, completion):
+        assert "bar.txt" in completion
+
+    @pytest.mark.complete("python -- foo ", cwd="python")
+    def test_script_args_3(self, completion):
+        assert "bar.txt" in completion
+
+    @pytest.mark.complete("python -- ", cwd="python")
+    def test_script_args_4(self, completion):
+        assert "bar.txt" not in completion
+
+    @pytest.mark.complete("python -W -mfoo ", cwd="python")
+    def test_script_args_5(self, completion):
+        assert "bar.txt" not in completion
```

Comment:
To clarify, this is because `-mfoo` gets treated as an arg to `-W`, and ignored as invalid -> this is effectively the same as calling just `python ` (as far as the python executable goes)? Perhaps reflecting that in the test function name would be in order, e.g. like this, or in its docstring? Pre-approved with something like that.

```suggestion
    @pytest.mark.complete("python -W -mfoo ", cwd="python")
    def test_script_invalid_w_arg(self, completion):
        assert "bar.txt" not in completion
```

Reply:
I've updated the function names and added a docstring to the last one.